apps/passport-client/components/screens/FrogScreens/GetFrogTab.tsx

```diff
@@ -0,0 +1,176 @@
+import { EdDSAFrogPCD, EdDSAFrogPCDPackage } from "@pcd/eddsa-frog-pcd";
+import { Subscription } from "@pcd/passport-interface";
+import { Separator } from "@pcd/passport-ui";
+import _ from "lodash";
+import prettyMilliseconds from "pretty-ms";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import styled from "styled-components";
+import { useDispatch, usePCDCollection } from "../../../src/appHooks";
+import { PCDCard } from "../../shared/PCDCard";
+import { ActionButton } from "./Button";
+
+export function GetFrogTab({
+  subs,
+  refetch
+}: {
+  subs: Subscription[];
+  refetch: () => Promise<void>;
+}) {
+  const pcds = usePCDCollection();
+  // NB: we cannot use useMemo because pcds are mutate in-place
+  const frogPCDs = pcds
+    .getAllPCDsInFolder("FrogCrypto")
+    .filter(
+      (pcd): pcd is EdDSAFrogPCD => pcd.type === EdDSAFrogPCDPackage.name
+    );
+
+  const [selectedPCDID, setSelectedPCDID] = useState("");
+  const selectedPCD = useMemo(
+    () => frogPCDs.find((pcd) => pcd.id === selectedPCDID) || frogPCDs[0],
+    [frogPCDs, selectedPCDID]
+  );
+  const onPcdClick = useCallback((id: string) => {
+    setSelectedPCDID(id);
+  }, []);
+
+  const [_lastestFrogPCD, setLatestFrogPCD] = useState<EdDSAFrogPCD>();
+  useEffect(() => {
+    const latest = _.maxBy(frogPCDs, (pcd) => pcd.claim.data.timestampSigned);
+    if (latest) {
+      setLatestFrogPCD((previous) => {
+        if (previous && previous.id !== latest.id) {
+          setSelectedPCDID(latest.id);
+        }
+
+        return latest;
+      });
```

Comment:
i'm noticing you're much more careful about avoiding extra re-renders than me. personally i would be ok w/ just replacing this block with 

```ts
setLatestFrogPCD(latest);
setSelectedPCDID(latest.id);
```